src/mqt/predictor/ml/predictor.py

```diff
@@ -199,7 +199,7 @@ def generate_compiled_circuits(
             delayed(self.compile_all_circuits_devicewise)(
                 device_name, timeout, figure_of_merit, source_path, target_path, logger.level
             )
-            for figure_of_merit in ["expected_fidelity", "critical_depth"]
+            for figure_of_merit in ["expected_fidelity", "critical_depth", "expected_success_probability"]
```

Comment:
Maybe we can iterate over `mqt.predictor.reward.figure_of_merit` instead of hard-coding the figures of merit?